sklearn/metrics/tests/test_pairwise.py

```diff
@@ -334,6 +350,53 @@ def test_pairwise_parallel(func, metric, kwds, dtype):
     assert_allclose(S, S2)
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name",
+    yield_namespace_device_dtype_combinations(),
+    ids=_get_namespace_device_dtype_ids,
+)
+@pytest.mark.parametrize(
+    "func, metric, kwds",
+    [
+        (pairwise_distances, "euclidean", {}),
+        (pairwise_kernels, "polynomial", {"degree": 1}),
+        (pairwise_kernels, callable_rbf_kernel, {"gamma": 0.1}),
+    ],
+)
+def test_pairwise_parallel_array_api(
+    func, metric, kwds, array_namespace, device, dtype_name
+):
+    xp = _array_api_for_tests(array_namespace, device)
+    rng = np.random.RandomState(0)
+    # Why 5 and not more? this seems to still result in a lot of 0 vaules?
+    X_np = np.array(5 * rng.random_sample((5, 4)), dtype=dtype_name)
```

Comment:
Not directly related to this PR, I think the `5 *` is to prevent lots of 0 values when dtype is `int` in the `test_pairwise_parallel` test above. While fiddling I noticed that even `5 *` results in a lot of 0 values, and wondered why not just increase it to e.g., 10?